From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: David Llewellyn-Jones <david@flypig.co.uk>
Date: Wed, 5 Aug 2026 11:26:40 +0200
Subject: [PATCH 37/67] Disable SessionStore functionality

Sailfish Browser performs session persistence outside Firefox SessionStore.

Keep the ESR140 XPCOM adapter contract but make both update methods no-ops.
---
 .../SessionStoreFunctions.sys.mjs             | 84 +------------------
 1 file changed, 2 insertions(+), 82 deletions(-)

diff --git a/browser/components/sessionstore/SessionStoreFunctions.sys.mjs b/browser/components/sessionstore/SessionStoreFunctions.sys.mjs
index 4e60d14135b8..1b2473262392 100644
--- a/browser/components/sessionstore/SessionStoreFunctions.sys.mjs
+++ b/browser/components/sessionstore/SessionStoreFunctions.sys.mjs
@@ -1,92 +1,12 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-import { SessionStore } from "resource:///modules/sessionstore/SessionStore.sys.mjs";
 
 export class SessionStoreFunctions {
-  UpdateSessionStore(
-    aBrowser,
-    aBrowsingContext,
-    aPermanentKey,
-    aEpoch,
-    aCollectSHistory,
-    aData
-  ) {
-    return SessionStoreFuncInternal.updateSessionStore(
-      aBrowser,
-      aBrowsingContext,
-      aPermanentKey,
-      aEpoch,
-      aCollectSHistory,
-      aData
-    );
-  }
-
-  UpdateSessionStoreForStorage(
-    aBrowser,
-    aBrowsingContext,
-    aPermanentKey,
-    aEpoch,
-    aData
-  ) {
-    return SessionStoreFuncInternal.updateSessionStoreForStorage(
-      aBrowser,
-      aBrowsingContext,
-      aPermanentKey,
-      aEpoch,
-      aData
-    );
-  }
+  UpdateSessionStore() {}
+  UpdateSessionStoreForStorage() {}
 }
 
-var SessionStoreFuncInternal = {
-  updateSessionStore: function SSF_updateSessionStore(
-    aBrowser,
-    aBrowsingContext,
-    aPermanentKey,
-    aEpoch,
-    aCollectSHistory,
-    aData
-  ) {
-    let { formdata, scroll } = aData;
-
-    if (formdata) {
-      aData.formdata = formdata.toJSON();
-    }
-
-    if (scroll) {
-      aData.scroll = scroll.toJSON();
-    }
-
-    SessionStore.updateSessionStoreFromTablistener(
-      aBrowser,
-      aBrowsingContext,
-      aPermanentKey,
-      {
-        data: aData,
-        epoch: aEpoch,
-        sHistoryNeeded: aCollectSHistory,
-      }
-    );
-  },
-
-  updateSessionStoreForStorage: function SSF_updateSessionStoreForStorage(
-    aBrowser,
-    aBrowsingContext,
-    aPermanentKey,
-    aEpoch,
-    aData
-  ) {
-    SessionStore.updateSessionStoreFromTablistener(
-      aBrowser,
-      aBrowsingContext,
-      aPermanentKey,
-      { data: { storage: aData }, epoch: aEpoch },
-      true
-    );
-  },
-};
-
 SessionStoreFunctions.prototype.QueryInterface = ChromeUtils.generateQI([
   "nsISessionStoreFunctions",
 ]);
